operator/consumer/queues_listener.go

```diff
@@ -68,23 +88,37 @@ func (l *QueuesListener) listen(ctx context.Context, rollupId uint32, rollupData
 				l.Remove(rollupId)
 				return
 			}
-
 			l.logger.Info("New delivery", "rollupId", rollupId)
 
 			publishPayload := new(PublishPayload)
 			err := borsh.Deserialize(publishPayload, d.Body)
 			if err != nil {
-				l.logger.Error("Error deserializing MQ payload")
+				l.logger.Fatal("Error deserializing payload")
 				continue
 			}
 
-			var block types.Block
-			if err := rlp.DecodeBytes(publishPayload.Data, &block); err != nil {
-				l.logger.Warn("Invalid block", "rollupId", rollupId, "err", err)
+			submitRequest := new(SubmitRequest)
+			err = borsh.Deserialize(submitRequest, publishPayload.Data)
+			if err != nil {
+				l.logger.Fatal("Invalid blob", "d.Body", d.Body, "err", err)
```

Comment:
Not sure if this should be fatal as well

Reply:
Fixed